### ISSUE TITLE
Setting for the 50k dataset

### ISSUE THREAD

**Issue by feiyang-cai** [yuqianghan/editretro#22]
Dear author, I tried to replicate the results for uspto50K dataset. 

Here is what I retrained (only finetuning not pretraining) based on your readme, but the result is much lower that what you reported in the paper.

<img width="876" alt="19d581b2ce211ba96230f538f1a81d6" src="https://github.com/user-attachments/assets/4e114c10-8e28-40f2-9fa9-fd7b7fe397df" />

I am thinking the problem might be from the setting of the generation... I saw in your default setting, you average the checkpoints from 1 to 40. Did you use the same setting in your experiments, or could you please share the setting you used for getting the reported results?
```
# python ./utils/average_checkpoints.py --inputs ${model_dir} \
#     --output ${ckpt_path} \
#     --num-epoch-checkpoints 40 \
# 	--checkpoint-upper-bound 40 \
```

**Comment by yuqianghan** [comment]
Dear,  based on the result you posted, I suspect that you may have adjusted some hyperparameters in the script. The predicted data size should be 5007, which corresponds to the size of the test set.

**Comment by feiyang-cai** [comment]
Thanks for your reply. 

I tried this again by 1. re-preprocessing the dataset; 2. re-evaluating the results. 

The results are still the same and the data size is 10014 (this is 2 x 5007). I just copied the scripts I used,  which is same as you provided (only change is to point my own checkpoint, I don't think the checkpoint can change the size of the dataset), so I don't think the problem is "I adjusted the hyperparameters in your script". 

```
#!/bin/bash

gpus="0"
aug=20
topk=20
beam_size=20
repos_beam=5
mask_beam=1
token_beam=4
max_tokens=2000   #!!!TODO: reduce this number if encountering CUDA OOM

databin=./datasets/USPTO_50K/aug20/data-bin  # binaried test data

#root_dir=results/finetune_50k/xxxxxxxx_xxxxxx  #TODO: point to the checkpoint path
root_dir=results/finetune_50k/20241223_200005
exp_n=2
outfile=generation
outputdir=${root_dir}/generations/${exp_n}
mkdir -p ${outputdir}

# If you are using our provided checkpoint, replace ckpt_path with the path to your specific checkpoint.
# ckpt_path=xxxxxx/checkpoint_50k.pt

####!!! If you fintune the model yourself, uncomment the following codes to process the checkpoints.
model_dir=${root_dir}/checkpoints
ckpt_name=finetune.pt
ckpt_path=${outputdir}/${ckpt_name}
####!!!TODO: average multiple (dozens of) checkpoints to get better performance
python ./utils/average_checkpoints.py --inputs ${model_dir} \
    --output ${ckpt_path} \
    --num-epoch-checkpoints 40 \
        --checkpoint-upper-bound 40 \

CUDA_VISIBLE_DEVICES=$gpus CUDA_LAUNCH_BLOCKING=1 fairseq-generate \
        --user-dir editretro \
        $databin \
        -s src -t tgt \
        --gen-subset test \
        --task translation_retro \
        --path ${ckpt_path} \
        --iter-decode-max-iter 10 \
        --iter-decode-eos-penalty 0 \
        --beam 1 --remove-bpe \
        --init-src \
        --TOPK ${beam_size} \
        --max-tokens ${max_tokens} \
        --repos-beam ${repos_beam} \
        --mask-beam ${mask_beam} \
        --token-beam ${token_beam} \
        --fp16 \
        --print-step --retain-iter-history >${outputdir}/${outfile}.txt \


# post processing
src=src.txt
tgt=tgt.txt
pred=pred.txt
prob=prob.txt
grep ^S ${outputdir}/${outfile}.txt | LC_ALL=C sort -V | cut -f2- > ${outputdir}/${src}
grep ^T ${outputdir}/${outfile}.txt | LC_ALL=C sort -V | cut -f2- > ${outputdir}/${tgt}
grep ^H ${outputdir}/${outfile}.txt | LC_ALL=C sort -V | cut -f3- > ${outputdir}/${pred}
grep ^P ${outputdir}/${outfile}.txt | LC_ALL=C sort -V | cut -f2- > ${outputdir}/${prob}


python ./utils/post_process.py \
    -generate_path  ${outputdir}/${pred} \
    -prob_path ${outputdir}/${prob} \
    -tgt_path ${outputdir}/${tgt} \
    -out_path ${outputdir}/${outfile}.json


# evaluate the results
python ./utils/score.py \
        -n_best ${topk} \
        -beam_size ${beam_size} \
        -predictions ${outputdir}/${outfile}.json \
        -targets ${outputdir}/${outfile}.json \
        -augmentation ${aug} \
        -score_alpha 0.1 \
```

**Comment by yuqianghan** [comment]
You can verify the size of the preprocessed dataset?

**Comment by feiyang-cai** [comment]
Checked. 5007.

**Comment by feiyang-cai** [comment]
Checked. 200280 (5007 x 40). And also I checked the pre-processed data (raw), it is 5007. 

<img width="398" alt="image" src="https://github.com/user-attachments/assets/71788720-0d04-4253-be01-6060b619bf83" />

**Comment by yuqianghan** [comment]
I figured out that I saved the pre-processed files using the 'a' mode
https://github.com/yuqianghan/editretro/blob/c2865b14f138614c35260391163968c6e7f223fb/preprocess/preprocess_data.py#L147
The size of pre-processed test data with _aug=20_ should be 100,140 (5007 X 20). Could you remove all previously processed files and regenerate them?

**Comment by feiyang-cai** [comment]
removed the previous data and re-preprocess the dataset again, but it is still the same thing. 

But I noticed that for each set, the data set is phrased twice. I am not sure if this is the problem.

<img width="960" alt="image" src="https://github.com/user-attachments/assets/4157f135-0706-4b7a-b3be-037c01b86c26" />

**Comment by yuqianghan** [comment]
Thank you for pointing out this issue. I carefully reviewed the preprocessing code and found that I had mixed the codes for preprocessing datasets with and without _batch_. I have updated the _preprocess_data.py_ file. It should now work correctly.

**Comment by feiyang-cai** [comment]
Great! I will try it now. BTW, did you average the checkpoint using 1-40 epoch for USPTO-50, or you use other setting?
```
python ./utils/average_checkpoints.py --inputs ${model_dir} \
    --output ${ckpt_path} \
    --num-epoch-checkpoints 40 \
        --checkpoint-upper-bound 40 \
```

**Comment by yuqianghan** [comment]
I used the latest 20 epoch checkpoints, but in my latter experience, using checkpoints from epochs 1 to 40 can achieve stable and better performance.

**Comment by feiyang-cai** [comment]
Sure, will try both. Thanks. Close the isssue

**Comment by feiyang-cai** [comment]
I used the new dataset to finetune and evaluate the results for 50k, but the results now are much worse. I am using the average checkpoint from 1-40 to do the evaluation. Do you have any idea why the results are so bad? 

<img width="788" alt="image" src="https://github.com/user-attachments/assets/032e8c9f-fa40-4be4-bc40-e0c986bd688d" />

**Comment by yuqianghan** [comment]
Could you evaluate the new dataset with our prepared checkpoint to compare the results? Then I can determine whether the issue stems from the dataset preprocessing or the fine-tuning process. https://drive.google.com/file/d/1nH9poV9FpO5nKcF1IMVZpeAACGeH94Gu/view?usp=drive_link
